request/src/main/java/com/vimeo/networking2/internal/VimeoApiClientImpl.kt

```diff
@@ -0,0 +1,835 @@
+            map[ApiConstants.Parameters.FILTER_DURATION] = searchDurationType.value
+                ?: error(INVALID_ENUM_MESSAGE)
+        }
+        if (searchFacetTypes != null) {
+            map[ApiConstants.Parameters.PARAMETER_GET_FACETS] =
+                searchFacetTypes.joinToString(separator = ",", transform = {
+                    it.value ?: error(INVALID_ENUM_MESSAGE)
+                })
+        }
+        if (category != null) {
+            map[ApiConstants.Parameters.FILTER_CATEGORY] = category
+        }
+        if (featuredVideoCount != null) {
+            map[ApiConstants.Parameters.FILTER_FEATURED_COUNT] = featuredVideoCount.toString()
+        }
+        if (containerFieldFilter != null) {
+            map[ApiConstants.Parameters.PARAMETER_GET_CONTAINER_FIELD_FILTER] = containerFieldFilter
+        }
+        if (queryParams != null) {
+            map.putAll(queryParams)
+        }
+
+        return vimeoService.search(authHeader, map).enqueue(callback)
+    }
+
+    override fun createPictureCollection(uri: String, callback: VimeoCallback<PictureCollection>): VimeoRequest {
+        return vimeoService.createPictureCollection(authHeader, uri).enqueue(callback)
+    }
+
+    override fun activatePictureCollection(uri: String, callback: VimeoCallback<PictureCollection>): VimeoRequest {
+        return vimeoService.editPictureCollection(
+            authHeader,
+            uri,
+            mapOf(ApiConstants.Parameters.PARAMETER_ACTIVE to true)
+        ).enqueue(callback)
+    }
+
+    override fun updateFollow(isFollowing: Boolean, uri: String, callback: VimeoCallback<Unit>): VimeoRequest {
+        return if (isFollowing) {
+            vimeoService.put(authHeader, uri, emptyMap()).enqueue(callback)
+        } else {
+            vimeoService.delete(authHeader, uri, emptyMap()).enqueue(callback)
+        }
+    }
+
+    override fun updateFollow(
+        isFollowing: Boolean,
+        followable: Followable,
+        callback: VimeoCallback<Unit>
+    ): VimeoRequest {
+        val uri = followable.metadata?.interactions?.follow?.uri
+            ?: return localVimeoCallAdapter.enqueueEmptyUri(callback)
+        return updateFollow(isFollowing, uri, callback)
+    }
+
+    override fun updateVideoLike(
+        isLiked: Boolean,
+        uri: String,
+        password: String?,
+        callback: VimeoCallback<Unit>
+    ): VimeoRequest {
+        val optionsMap = password.asPasswordParameter()
+        return if (isLiked) {
+            vimeoService.put(authHeader, uri, optionsMap).enqueue(callback)
+        } else {
+            vimeoService.delete(authHeader, uri, optionsMap).enqueue(callback)
+        }
+    }
+
+    override fun updateVideoLike(
+        isLiked: Boolean,
+        video: Video,
+        password: String?,
+        callback: VimeoCallback<Unit>
+    ): VimeoRequest {
+        val uri = video.uri ?: return localVimeoCallAdapter.enqueueEmptyUri(callback)
```

Comment:
Why just checking against non-null and not non-blank?

Reply:
Added in a `.notEmpty()` extension that filters out the URI if it is null or blank.